From 22fb8b2690a5fa364d252846f06b77b5fec8c602 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Thu, 7 Jan 2021 10:27:43 -0500
Subject: [PATCH 3/6] Fix cherry-pick erorr

---
 ldap/servers/slapd/slapi-private.h | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/ldap/servers/slapd/slapi-private.h b/ldap/servers/slapd/slapi-private.h
index e0092d571..d5abe8ac1 100644
--- a/ldap/servers/slapd/slapi-private.h
+++ b/ldap/servers/slapd/slapi-private.h
@@ -1476,17 +1476,6 @@ typedef enum task_warning_t{
 int32_t slapi_pblock_get_task_warning(Slapi_PBlock *pb);
 void slapi_pblock_set_task_warning(Slapi_PBlock *pb, task_warning warn);
 
-/* task warnings */
-typedef enum task_warning_t{
-    WARN_UPGARDE_DN_FORMAT_ALL    = (1 << 0),
-    WARN_UPGRADE_DN_FORMAT        = (1 << 1),
-    WARN_UPGRADE_DN_FORMAT_SPACE  = (1 << 2),
-    WARN_SKIPPED_IMPORT_ENTRY     = (1 << 3)
-} task_warning;
-
-int32_t slapi_pblock_get_task_warning(Slapi_PBlock *pb);
-void slapi_pblock_set_task_warning(Slapi_PBlock *pb, task_warning warn);
-
 int slapi_exists_or_add_internal(Slapi_DN *dn, const char *filter, const char *entry, const char *modifier_name);
 
 #ifdef __cplusplus
-- 
2.26.2

